node/src/main.rs

```diff
@@ -63,38 +63,45 @@ lazy_static! {
 git_testament!(TESTAMENT);
 
 fn main() {
-    let (shutdown_sender, shutdown_receiver) = oneshot::channel();
-    // Register guarded panic logger which ensures logs flush on shutdown
-    let (panic_logger, _panic_guard) = guarded_logger();
-    register_panic_hook(panic_logger, shutdown_sender);
-
-    // Create components for tokio context: multi-threaded runtime,
-    // executor context on the runtime, and Timer handle.
-    let runtime = tokio::runtime::Runtime::new().expect("Failed to create runtime");
-    let mut executor = runtime.executor();
+    use std::sync::Mutex;
+    use tokio::runtime;
+
+    // Create components for tokio context: multi-threaded runtime, executor
+    // context on the runtime, and Timer handle.
+    //
+    // Configure the runtime to shutdown after a panic.
+    let runtime: Arc<Mutex<Option<runtime::Runtime>>> = Arc::new(Mutex::new(None));
+    let handler_runtime = runtime.clone();
+    *runtime.lock().unwrap() = Some(
+        runtime::Builder::new()
+            .panic_handler(move |_| {
+                let runtime = handler_runtime.clone();
+                std::thread::spawn(move || {
+                    if let Some(runtime) = runtime.lock().unwrap().take() {
+                        // Try to cleanly shutdown the runtime, but
+                        // unconditionally exit after a while.
+                        std::thread::spawn(|| {
+                            std::thread::sleep(Duration::from_millis(3000));
+                            std::process::exit(1);
+                        });
+                        runtime
+                            .shutdown_now()
+                            .wait()
+                            .expect("Failed to shutdown Tokio Runtime");
+                        println!("Runtime cleaned up and shutdown successfully");
+                    }
+                });
+            })
```

Comment:
It looks like the old panic handler tried to log details about the panic, and the new one doesn't. Shouldn't that information still be logged?

Reply:
Good point. This removes the custom panic handler and uses the default panic handler, which will print the panic information to stdout. I'm assuming this is good enough, and that we don't need to have the panic information go through `slog`.

Reply:
Makes sense to me.